python/chronos/src/bigdl/chronos/forecaster/base_forecaster.py

```diff
@@ -1231,6 +1300,28 @@ def quantize(self, calib_data=None,
         if approach == 'dynamic' and calib_data is not None:
             invalidInputError(False, "You must not set a `calib_data` for dynamic quantization.")
 
+        # change tsdataset to dataloader
+        if isinstance(calib_data, TSDataset):
+            _rolled = calib_data.numpy_x is None
+            calib_data = calib_data.to_torch_data_loader(
+                batch_size=1,
+                roll=_rolled,
+                lookback=self.data_config['past_seq_len'],
+                horizon=self.data_config['future_seq_len'],
+                feature_col=calib_data.roll_feature,
+                target_col=calib_data.roll_target,
+                shuffle=False)
+        if isinstance(val_data, TSDataset):
+            _rolled = val_data.numpy_x is None
+            val_data = val_data.to_torch_data_loader(
+                batch_size=1,
+                roll=_rolled,
+                lookback=self.data_config['past_seq_len'],
+                horizon=self.data_config['future_seq_len'],
+                feature_col=val_data.roll_feature,
+                target_col=val_data.roll_target,
+                shuffle=False)
```

Comment:
later (not in this PR, we may need a utils func for this)

Reply:
will consider this.